src/solana/index.ts

```diff
@@ -0,0 +1,378 @@
+   * @returns A promise that resolves to a VersionedTransaction object.
+   */
+  public async prepareTransaction(
+    instructions: TransactionInstruction[],
+    payer: PublicKey,
+  ): Promise<VersionedTransaction> {
+    const blockhash = await this.getBlockhash();
+    const messageV0 = new TransactionMessage({
+      payerKey: payer,
+      recentBlockhash: blockhash,
+      instructions,
+    }).compileToV0Message();
+
+    return new VersionedTransaction(messageV0);
+  }
+  /**
+   * Retrieves the latest blockhash.
+   *
+   * @returns A promise that resolves to a string representing the blockhash.
+   */
+  public async getBlockhash(): Promise<string> {
+    const res = await this.connection.getLatestBlockhash({
+      commitment: 'max',
+    });
+
+    return res.blockhash;
+  }
+
+  /**
+   * Delegates a specified amount from a stake account to a validator.
+   *
+   * @param token - The token to be used for the delegation.
+   * @param address - The public key of the account.
+   * @param amount - The amount to be delegated.
+   * @param stakeAccount - The public key of the stake account.
+   * @throws Throws an error if the token is invalid, the amount is less than the minimum amount, or if there's an issue during the delegation process.
+   * @returns Returns a promise that resolves with the delegation transaction.
+   *
+   */
+  public async delegate(
+    token: string,
+    address: string,
+    amount: number,
+    stakeAccount: string,
+  ): Promise<ApiResponse<VersionedTransaction>> {
+    const isTokenValid = await CheckToken(token);
+    if (!isTokenValid) {
+      this.throwError('INVALID_TOKEN_ERROR');
+    }
+
+    if (amount < MIN_AMOUNT) {
+      this.throwError('MIN_AMOUNT_ERROR', MIN_AMOUNT.toString());
+    }
+
+    try {
+      const publicKey = new PublicKey(address);
+      const stakeAccountPublicKey = new PublicKey(stakeAccount);
+      const selectedValidatorPubkey = new PublicKey(VALIDATOR_ADDRESS);
+
+      const delegateTx = new Transaction().add(
+        StakeProgram.delegate({
+          stakePubkey: stakeAccountPublicKey,
+          authorizedPubkey: publicKey,
+          votePubkey: selectedValidatorPubkey,
+        }),
+      );
+
+      const delegateVerTx = await this.prepareTransaction(
+        delegateTx.instructions,
+        publicKey,
+      );
+
+      await SetStats({
+        token,
+        action: 'stake',
+        amount,
+        address,
+        chain: CHAIN,
+      });
+
+      return { result: delegateVerTx };
+    } catch (error) {
+      throw this.handleError('DELEGATE_ERROR', error);
+    }
+  }
+
+  /**
+   * Deactivates a stake account.
+   *
+   * @param address - The public key of the account.
+   * @param stakeAccountPublicKey - The public key of the stake account.
+   * @throws Throws an error if there's an issue during the deactivation process.
+   * @returns Returns a promise that resolves with the deactivation transaction.
+   *
+   */
+  public async deactivate(
```

Comment:
Why do some methods check the API token and some don't? Maybe it should be checked here as well?